adagrams/game.py

```diff
@@ -1,11 +1,180 @@
+        idx = letters.index(letter)
+        # Letter with weight 0 should have no chance of being drawn
+        assert weights[idx] > 0
+        hand.append(letter)
+        weights[idx] -= 1
+
+    return hand
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    """
+    Returns a boolean.
+    Checks if an input word is valid based on available letters in hand (i.e., letter_bank).
+    Does not mutate the hand.
+    Allows for lowercase letters.
+    """
+
+    copy_bank = [a.upper() for a in letter_bank]
+    word = word.upper()
+
+    for char in word:
+        if char not in copy_bank:
+            return False
+        copy_bank.remove(char)
+
+    return True
+
 
 def score_word(word):
-    pass
+    """
+    Returns an integer representing the number of points.
+    A word with length greater than 6 gets an additional 8 points.
+    """
+    word = word.upper()
+
+    score = sum(SCORE_CHART[char] for char in word)
+    
+    # Give bonus for words longer than 6 letters
+    if len(word) > 6:
+        score += 8
+
+    return score
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    """
+    Returns a tuple of the winning word and its score.
+    In case of a tie, prioritize the word with 10 letters, 
+    then the word with the fewest letters, then the first element in the list
+    """
+    
+    word_scores = []
+    max_score = 0
+    
+    for idx, word in enumerate(word_list):
+        score = score_word(word)
+        if score > max_score:
+            max_score = score
+        word_scores.append((word, score, idx))
+
+    # Sort word_scores in descending order of scores
+    word_scores.sort(reverse=True, key=lambda x: x[1])
+  
+    # Create list of words and their indices with tied scores
+    ties = [(word, idx) for word, score, idx in word_scores if score == max_score]
+
+    # Handle the case of a clear winner
+    if len(ties) == 1:
+        winner = ties[0][0]
+        return winner, max_score
+
+    # Sort list of ties in ascending order of word length
+    ties.sort(key=lambda x:len(x[0]))
+
+    # Prioritize words with length 10 as winners
+    if len(ties[-1][0]) == 10:
+        # Mutate ties to only contain words with length 10
+        ties = [(word, idx) for word, idx in ties if len(word) == 10]
+        # Select the word with lowest index if there's more than one option for winner
+        if len(ties) != 1:
+            ties.sort(key=lambda x:x[1])
+        winner = ties[0][0]
+    
+    # Secondary win condition: words with fewest letters
+    else:
+        min_length = len(ties[0][0])
+        # Mutate ties to only contain words with fewest letters
+        ties = [(word, idx) for word, idx in ties if len(word) == min_length]
+        # Select the word with lowest index if there's more than one option for winner
+        if len(ties) != 1:
+            ties.sort(key=lambda x:x[1])
+        winner = ties[0][0]
```

Comment:
While on a technical level, this is very creative, and shows good research and understanding, it's hard to read. It's a fine line between concise, elegant solutions and being understandable to humans with a glance or two.

Consider how you might refactor the ties under one for loop as you calculate the winner. 

But, great job! I can tell you had a lot of fun working with these methods!